### PR TITLE
shaderc: fix glslc on aarch64-darwin

### LINKED ISSUE #186148: shacerc: glslc gets killed on aarch64-darwin (missing/invalid signature)
### Describe the bug
glslc binary in shaderc has invalid signature

### Steps To Reproduce
1. try to compile shader on aarch64-darwin to SPIR-V

### Expected behavior
`glslc` should not be killed

### Additional context
```
$ codesign --verify /nix/store/bh7c8yr9f8d63insp94x9721ppgnjr6b-shaderc-2021.0-bin/bin/glslc                                                                                                                                   main!?
/nix/store/bh7c8yr9f8d63insp94x9721ppgnjr6b-shaderc-2021.0-bin/bin/glslc: invalid signature (code or signature have been modified)
In architecture: arm64
```
### Metadata
`nixpkgs/dd37f38d62587f42b8beb700a2ac98d92ac22de2#legacyPackages.aarch64-darwin.shaderc`

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

Closes #186148.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/shaderc/default.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchFromGitHub, cmake, darwin, python3 }:
+{ lib, stdenv, fetchFromGitHub, cmake, darwin, python3, autoSignDarwinBinariesHook }:
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/shaderc/default.nix:1]:
```suggestion
{ lib, stdenv, fetchFromGitHub, cmake, cctools, python3, autoSignDarwinBinariesHook }:
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -26375,7 +26375,9 @@ with pkgs;
 
   session-desktop = callPackage ../applications/networking/instant-messengers/session-desktop { };
 
-  shaderc = callPackage ../development/compilers/shaderc { };
+  shaderc = callPackage ../development/compilers/shaderc {
+    inherit (darwin) autoSignDarwinBinariesHook;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:26379]:
```suggestion
    inherit (darwin) autoSignDarwinBinariesHook cctools;
```

File: pkgs/development/compilers/shaderc/default.nix [hunk 3]
```diff
@@ -42,7 +43,9 @@ in stdenv.mkDerivation rec {
     ln -s ${spirv-headers} third_party/spirv-tools/external/spirv-headers
   '';
 
-  nativeBuildInputs = [ cmake python3 ] ++ lib.optionals stdenv.isDarwin [ darwin.cctools ];
+  nativeBuildInputs = [ cmake python3 ]
+    ++ lib.optionals stdenv.isDarwin [ darwin.cctools ]
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/shaderc/default.nix:47]:
```suggestion
    ++ lib.optionals stdenv.isDarwin [ cctools ]
```

File: pkgs/development/compilers/shaderc/default.nix (hunk 1 above)
Comment (wegank) [reply pkgs/development/compilers/shaderc/default.nix]: Done.

**Pull request merged**